app/services/metrc_service/plant/move.rb

```diff
@@ -18,36 +17,25 @@ def transaction
       end
 
       def prior_move
-        return @prior_move if @prior_move
-
-        previous_move = Transaction.where(
-          'batch_id = ? AND type = ? AND vendor = ? AND id NOT IN (?)',
-          @batch_id,
-          :move_batch,
-          :metrc,
-          transaction.id
-        ).limit(1).order('created_at desc').first
+        previous_move = batch.completions.select do |comp|
+          comp.action_type == 'move' && comp.id < @completion_id
+        end.max_by(&:start_time)
```

Comment:
completions often share the same start time, so you need to also sort by `id` as a tie-breaker

Reply:
I think in this case, it would be rare to have two moves at the same time, so it's ok... but in general it's always good to remember that sorting only by one field is usually a source of unexpected behavior if it's not a unique field

Reply:
mmmm maybe max_by id then?

Reply:
✅